### PR TITLE
Some issues fixing for android and other linux OSs

### PR DESCRIPTION
**Pull request by ssrlive**
Some issues fixing for android and other linux OSs and extract the ANDROID_APP_PACKAGE_NAME as a macro definition.

### CODE REVIEW

File: tun2socks/tun2socks.c [hunk 1]
```diff
@@ -297,9 +301,9 @@ LinkedList1 tcp_clients;
 // number of clients
 int num_clients;
 
-#ifdef __ANDROID__
 // Address of dnsgw
 BAddr dnsgw;
+#ifdef __ANDROID__
```
Comment (Mygod) [inline tun2socks/tun2socks.c:302]: Why move this?

File: tun2socks/tun2socks.c [hunk 2]
```diff
@@ -485,11 +489,13 @@ int main (int argc, char **argv)
...
 
+#ifdef __ANDROID__
     if (options.fake_proc) {
         // Fake process name to cheat on Lollipop
-        strcpy(argv[0], "com.github.shadowsocks");
-        prctl(PR_SET_NAME, "com.github.shadowsocks");
+        strcpy(argv[0], ANDROID_APP_PACKAGE_NAME);
+        prctl(PR_SET_NAME, ANDROID_APP_PACKAGE_NAME);
     }
+#endif
```
Comment (Mygod) [inline tun2socks/tun2socks.c]: I believe neither of these are being used by shadowsocks-android latest master.

File: tun2socks/tun2socks.c (hunk 1 above)
Comment (ssrlive) [reply tun2socks/tun2socks.c:302]: in this line, `dnsgw` is used. 
https://github.com/shadowsocks/badvpn/blob/fb01eb983915c9a09a690ad44c6028dd87500ec7/tun2socks/tun2socks.c#L633

File: tun2socks/tun2socks.c (hunk 2 above)
Comment (ssrlive) [reply tun2socks/tun2socks.c]: I have tested it in Linux, it will error if not check `__ANDROID__`.

Comment (Mygod) [reply tun2socks/tun2socks.c]: I was talking about `com.github.shadowsocks`. Maybe it is appropriate to remove these two altogether.

**Pull request merged**